modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -1132,6 +1322,87 @@ private Folder _getFolder(NotificationQueueEntry notificationQueueEntry)
 				notificationQueueEntry.getNotificationQueueEntryId()));
 	}
 
+	private Set<String> _getTermNames(
+			long objectDefinitionId, Set<String> selectedTermCategories)
+		throws Exception {
+
+		Set<String> termNames = new HashSet<>();
+
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		mockLiferayResourceRequest.addParameter(
+			"objectDefinitionId", String.valueOf(objectDefinitionId));
+
+		MockLiferayResourceResponse mockLiferayResourceResponse =
+			new MockLiferayResourceResponse();
+
+		_mvcResourceCommand.serveResource(
+			mockLiferayResourceRequest, mockLiferayResourceResponse);
+
+		ByteArrayOutputStream byteArrayOutputStream =
+			(ByteArrayOutputStream)
+				mockLiferayResourceResponse.getPortletOutputStream();
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(
+			byteArrayOutputStream.toString());
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			if (!selectedTermCategories.contains(
+					jsonObject.getString("label"))) {
+
+				continue;
+			}
+
+			JSONArray itemsJSONArray = jsonObject.getJSONArray("items");
+
+			for (int j = 0; j < itemsJSONArray.length(); j++) {
+				JSONObject termJSONObject = itemsJSONArray.getJSONObject(j);
+
+				if (termJSONObject == null) {
+					continue;
+				}
+
+				termNames.add(termJSONObject.getString("content"));
+			}
+		}
+
+		return termNames;
+	}
+
+	private void _pushServiceContext() throws PortalException {
+		HttpServletRequest httpServletRequest =
+			ObjectActionThreadLocal.getHttpServletRequest();
+
+		if (httpServletRequest == null) {
+			httpServletRequest = new MockHttpServletRequest(
+				null, StringPool.BLANK, RandomTestUtil.randomString());
+
+			ThemeDisplay themeDisplay = new ThemeDisplay();
+
+			themeDisplay.setLocale(LocaleUtil.US);
+
+			httpServletRequest.setAttribute(
+				WebKeys.THEME_DISPLAY, themeDisplay);
+
+			ObjectActionThreadLocal.setHttpServletRequest(httpServletRequest);
+		}
```

Comment:
Hey @devpedrotavares it seems that the request retrieved at ObjectActionThreadLocal.getHttpServletRequest(); always will be null at this another [point](https://github.com/liferay-objects/liferay-portal/pull/3606/files#diff-eca215044a4eb53b9507b221f69a273025af4136f243a39d4b6c01cdccb5dcf8R1261), is that right?

If so, could you please refactor this method removing this conditional and assuming that the request have to be mocked whenever this method is called. 

Thank you!

Reply:
Okay!